generals/core/replay.py

```diff
@@ -33,31 +34,36 @@ def load(cls, path):
     def play(self):
         agents = [agent for agent in self.agent_data.keys()]
         game = Game(self.grid, agents)
-        gui = GUI(game, self.agent_data, from_replay=True)
+        gui = GUI(game, self.agent_data, mode="replay")
         gui_properties = gui.properties
 
         game_step, last_input_time, last_move_time = 0, 0, 0
         while 1:
             _t = time.time()
             # Check inputs
             if _t - last_input_time > 0.008:  # check for input every 8ms
-                control_events = gui.tick()
+                command = gui.tick()
                 last_input_time = _t
             else:
-                control_events = {"time_change": 0}
-            if "restart" in control_events:
+                command = ReplayCommand()
+            if command.quit:
+                import pygame
```

Comment:
This should be part of GUI (`close` method).
It makes more sense logically, and you also avoid copying code here and in `Gym/PZ integrations`.
Maybe even the `if` check?

Reply:
I think you are right, I'll quit pygame in GUI `close` method, but will still pass the signal to `integrations`, because according to APIs, there should be `close` function implemented

Reply:
Which can just call `self.gui.close()` if gui is already created btw.

Reply:
I believe this is fixed now.

Reply:
I think this is now resolved.
When game is requested to `quit` in replay mode, it is terminated with `quit()`. However, `self.gui.close()` is still available, because user might want to quit rendering programatically (see [example](https://gymnasium.farama.org/) with `env.close()`)